internal/expr/prepare.go

```diff
@@ -77,33 +78,67 @@ func starCheckOutput(p *outputPart) error {
 }
 
 // prepareInput checks that the input expression corresponds to a known type.
-func prepareInput(ti typeNameToInfo, p *inputPart) (typeMember, error) {
+func prepareInput(ti typeNameToInfo, p *inputPart) (*inputChunk, []typeMember, error) {
 	info, ok := ti[p.source.prefix]
 	if !ok {
 		ts := getKeys(ti)
 		if len(ts) == 0 {
-			return nil, fmt.Errorf(`type %q not passed as a parameter`, p.source.prefix)
+			return nil, nil, fmt.Errorf(`type %q not passed as a parameter`, p.source.prefix)
 		} else {
-			return nil, fmt.Errorf(`type %q not passed as a parameter, have: %s`, p.source.prefix, strings.Join(ts, ", "))
+			return nil, nil, fmt.Errorf(`type %q not passed as a parameter, have: %s`, p.source.prefix, strings.Join(ts, ", "))
 		}
 	}
 	switch info := info.(type) {
 	case *mapInfo:
-		return mapKey{name: p.source.name, mapType: info.typ()}, nil
+		return &inputChunk{}, []typeMember{&mapKey{name: p.source.name, mapType: info.typ()}}, nil
 	case *structInfo:
 		f, ok := info.tagToField[p.source.name]
 		if !ok {
-			return nil, fmt.Errorf(`type %q has no %q db tag`, info.typ().Name(), p.source.name)
+			return nil, nil, fmt.Errorf(`type %q has no %q db tag`, info.typ().Name(), p.source.name)
 		}
-		return f, nil
+		return &inputChunk{}, []typeMember{f}, nil
 	default:
-		return nil, fmt.Errorf(`internal error: unknown info type: %T`, info)
+		return nil, nil, fmt.Errorf(`internal error: unknown info type: %T`, info)
 	}
 }
 
+// prepareIn generates the values to fetch for sqlair input expressions following an IN statement.
+func prepareIn(ti typeNameToInfo, p *inPart) (*inChunk, []typeMember, error) {
```

Comment:
Now that there are `prepareIn`, `prepareInput` and `prepareOutput`, perhaps it would be cleaner if these were refactored into methods? Perhaps a common interface could then be used to simplify where these methods would be called

Reply:
True, it would be a good thing to do, though I might leave it for a future PR, theres probably going to be quite a few others that get in before this one that touch these methods so having them in a different place will make merging much more of a pain than it needs to be. But certainly something to do